### PR TITLE
Critic input normalization

### PR DESCRIPTION
**Pull request by JasperHoffmann**
We added some improvements to all sac implementations and sac_fop:
- Normalizing actions for the critic input for sac, sac_zop and sac_fop
- Made the masking of samples stricter. The mpc needs to converge for the current action and next state action.
- Added three example scripts for the quadrotor task.
- Removed unnecessary code in `leap_c/nn/utils.py`

### CODE REVIEW

File: leap_c/rl/sac_fop.py [hunk 1]
```diff
@@ -242,6 +255,25 @@ def train_loop(self) -> Iterator[int]:
 
                 # sample action
                 pi_o = self.pi(o, ps_sol)
+                with torch.no_grad():
+                    pi_o_prime = self.pi(o_prime, ps_sol)
+
+                pi_o_stats = pi_o.stats
+
+                # compute mask
+                mask_status = (pi_o.status == 0) & (pi_o_prime.status == 0)
```
Comment (dirkpr) [inline leap_c/rl/sac_fop.py:264]: Should we define a mask somewhere else and configure the acceptable status here? Max iterations is not status 0 for instance.

File: leap_c/nn/utils.py [hunk 2]
```diff
@@ -70,33 +18,36 @@ def string_to_activation(activation: str) -> nn.Module:
...
-        fcs.extend([nn.Linear(prev_d, d), string_to_activation(activation)])
-    if use_activation_on_output:
-        mlp = nn.Sequential(*fcs)
-    else:
-        mlp = nn.Sequential(*fcs[:-1])
 
-    # TODO(Leonard): Do we need orthogonal initialization?
-    # mlp.apply(weight_init)
-    return mlp
+def normalize(
```
Comment (dirkpr) [inline leap_c/nn/utils.py:34]: min_max_scaling would be more specific, but I guess this is accepted ML lingo

**Review by dirkpr — APPROVED** [review]
Looks good. Only two minor comments.

File: leap_c/nn/utils.py (hunk 2 above)
Comment (JasperHoffmann) [reply leap_c/nn/utils.py:34]: I think this a good idea :)

File: leap_c/rl/sac_fop.py (hunk 1 above)
Comment (JasperHoffmann) [reply leap_c/rl/sac_fop.py:264]: Yes, sounds great, we can also even allow in the config to determine, which stati should be filtered out :)

Comment (Confectio) [reply leap_c/rl/sac_fop.py:264]: But maybe we need to make it even more general than just specifying which status are ok? E.g., max iterations can mean the sum of residuals still being >> 1 (not ok?), but it could also mean the sum of residuals being at 1e-4 (ok?).

Comment (dirkpr) [reply leap_c/rl/sac_fop.py:264]: I would just mask for status at this point and have user specify acceptable solver tolerance in the configure_ocp_solver function

Comment (JasperHoffmann) [reply leap_c/rl/sac_fop.py:264]: Yes, I think this easies to implement as well :)

Comment (JasperHoffmann) [reply leap_c/rl/sac_fop.py:264]: I decided against implementing it. I think this is an Acados Specific detail that should not be handled in the sac but in the MpcLayer :)
